File: reporter.py
# coding: utf-8


class Reporter(object):
    def __init__(self, name):
        self.name = name
        self.cnt = 0
        self.cliques = []

    def inc_count(self):
        self.cnt += 1

    def record(self, clique):
        self.cliques.append(clique)

    def print_report(self):
        print(self.name)
        print(f'{self.cnt} recursive calls')
        for i, clique in enumerate(self.cliques):
            print(f'{i}: {clique}')
        print

File: go/src/bron_kerbosch/reporter.go
package bron_kerbosch

type Reporter interface {
	Record(clique []Vertex)
}

type SimpleReporter struct {
	cliques [][]Vertex
}

func (r *SimpleReporter) Record(clique []Vertex) {
	cc := make([]Vertex, len(clique))
	copy(cc, clique)
	r.cliques = append(r.cliques, cc)
}
